// docz-theme-umi/es/components/shared/Sidebar/CustomMenu.js
import React from 'react';
import { Link, useDocs } from 'docz';
import styled from 'styled-components';
var MenuContainer = styled.div.withConfig({
  displayName: "CustomMenu__MenuContainer"
})(["width:100%;padding:20px 0 24px 24px;line-height:22px;overflow-y:auto;flex:1;"]);
var Menu = styled.div.withConfig({
  displayName: "CustomMenu__Menu"
})(["width:100%;"]);
var ItemTitle = styled.div.withConfig({
  displayName: "CustomMenu__ItemTitle"
})(["color:rgba(0,0,0,0.45);margin-bottom:24px;font-size:14px;"]);
var Item = styled.div.withConfig({
  displayName: "CustomMenu__Item"
})(["cursor:pointer;color:rgba(0,0,0,0.65);margin-bottom:24px;transition:color 0.3s;:hover{> a{color:#1890ff;}}> a{display:block;font-size:", ";line-height:", ";color:", ";}"], function (props) {
  return props.size === 'large' ? '14px' : '12px';
}, function (props) {
  return props.size === 'large' ? '22px' : '20px';
}, function (p) {
  return p.selected ? '#1890ff' : 'rgba(0, 0, 0, 0.65)';
});

var CustomMenu = function CustomMenu(_ref) {
  var query = _ref.query;
  var _location = location,
      pathname = _location.pathname;
  var docs = useDocs(); // 按 menu 组织文档顺序

  var docsArrangedInMenu = docs && docs.filter(function (v) {
    return v.name.indexOf(query) > -1;
  }).reduce(function (list, doc) {
    if (doc.menu) {
      // 如果配置了 menu，将同一 menu 下的所有 docs 放到一个 SubMenu 内展示
      var existedMenu = list.find(function (v) {
        return v.name === doc.menu;
      });

      if (existedMenu) {
        var sortedDocs = existedMenu.menu.concat([doc]).sort(function (a, b) {
          return a.order - b.order;
        });
        existedMenu.menu = sortedDocs;
      } else {
        list.push({
          name: doc.menu,
          menu: [doc],
          menuOrder: doc.menuOrder || doc.order
        });
      }
    } else {
      // 如果没有配置 menu 属性，直接当作 Menu.Item 处理
      list.push({
        name: '',
        menu: [doc],
        menuOrder: doc.menuOrder || doc.order
      });
    }

    return list;
  }, []).sort(function (a, b) {
    return a.menuOrder - b.menuOrder;
  });
  return React.createElement(MenuContainer, null, React.createElement(Menu, null, docsArrangedInMenu && docsArrangedInMenu.map(function (item) {
    return item.name ? React.createElement(React.Fragment, {
      key: item.name
    }, React.createElement(ItemTitle, null, item.name), item.menu && item.menu.map(function (v) {
      return React.createElement(Item, {
        selected: v.route === pathname,
        key: v.route
      }, React.createElement(Link, {
        to: v.route
      }, React.createElement("span", null, v.name)));
    })) : React.createElement(React.Fragment, {
      key: item.menu[0].route
    }, React.createElement(Item, {
      selected: item.menu[0].route === pathname,
      size: "large"
    }, React.createElement(Link, {
      to: item.menu[0].route
    }, React.createElement("span", null, item.menu[0].name))));
  })));
};

export default CustomMenu;
